### DISCUSSION TITLE
Glitchy sound issues

### DISCUSSION THREAD

**Discussion by lukemosse** [CoderLine/alphaTab#1255] [General]
Hi there

I'm loving alphatab and have been developing a little jazz training app with it.

One question though - under certain situations it can kind of freeze and the audio gets stuck in a glitch/buzz.

Is there a way to avoid this happening?

Thanks

Luke

**Comment by Danielku15** [comment]
Can you share a bit more information about your setup? Audio playback glitches are mainly a result of bad performance. And there are many factors influencing this.

**Comment by lukemosse** [comment]
Thanks Daniel.

I'm just running it through a browser - my computer itself is relatively
high spec - I'm an After Effects animator so I have a fast processor and
lots of RAM etc.

There's a slight possibility that querying the currentTime value is causing
an issue as I have a scrolling div and another value triggered by
currentTime changing.

**Reply by Danielku15** [reply]
I don't even have problems with performance on some low spec android devices. What is your concrete test setup (hardware specs, browser, test application,..)?

**Comment by lukemosse** [comment]
I think in that case it must be the frequency with which I'm calling the
event listener. Chrome, my own application I'm working on, and 32gb ram,
ryzen 5 3600 6- core 3.8ghz - I don't think it will be the machine.

I sent the app through to a friend to test and it glitched for him too. I
think it must be something I'm doing with the event listeners that's just
too heavy for the browser.

**Reply by Danielku15** [reply]
Generally browsers have good profiling tools which might help you narrowing down the problem. If you can share some code or even a sample project I could have a look.

**Reply by lukemosse** [reply]
The project is here but it requires google sign in - https://phraseology-guitar-dev.web.app/
